stop plotting uninitialized memory

dhroth opened the pull request:
The main change is commit bb73e09: histograms were plotting uninitialized garbage from memory because the mask was being ignored. This sometimes shows up as a spike at the value 0 because uninitialized memory often contains zero. In addition, the values in the histograms were sometimes artificially increased if some bit of memory happened to have the right values. I explicitly exclude the masked values to fix this problem.

Other minor fixes are changing sims.utils.Site.elev to .height and removing hard-coded coolumn names in hourglassMetric.

rhiannonlynne commented on python/lsst/sims/maf/db/opsimDatabase.py:
```diff
@@ -154,7 +154,7 @@ def fetchLatLonHeight(self):
             lat = site.latitude_rad
             lon = site.longitude_rad
-            height = site.elev
+            height = site.height
```
why was this even working before?? gah. Thanks.

dhroth replied:
gotta love interpreted languages that only raise errors on code that actually gets run =)

rhiannonlynne commented on python/lsst/sims/maf/metrics/hourglassMetric.py:
```diff
@@ -35,7 +35,7 @@ def run(self, dataSlice, slicePoint=None):
         types = ['float']*len(names)
         pernight = np.zeros(len(unights), dtype=list(zip(names, types)))
-        pernight['mjd'] = dataSlice['observationStartMJD'][uindx]
+        pernight['mjd'] = dataSlice[self.mjdcol][uindx]
```
I don't understand what's going on here. In LSST master, this is already fixed. (and we'd like to use mjdCol to match LSST's (old) standard for camelcase). 
Same wither filterCol and nightCol.

rhiannonlynne commented on python/lsst/sims/maf/plots/spatialPlotters.py:
```diff
@@ -230,7 +230,7 @@ def __call__(self, metricValue, slicer, userPlotDict, fignum=None):
         if plotDict['scale'] is None:
             plotDict['scale'] = (hp.nside2pixarea(slicer.nside, degrees=True) / 1000.0)
-        fignum = self.baseHist(metricValue, slicer, plotDict, fignum=fignum)
+        fignum = self.baseHist(metricValue[~metricValue.mask], slicer, plotDict, fignum=fignum)
```
This is in the histogram section? This is also already fixed, by using only the compressed metricValue in the baseHist itself (so the full metricValue masked array should still be sent, otherwise things will fail in baseHist).

dhroth commented:
Hi Lynne, I think the hourglass problem is due to my git incompetence -- I had removed the hard-coded column names in my local repo a while ago and then tried to overwrite my local changes with master once it was fixed remotely, but I must have done it incorrectly.

I probably won't have a chance to test out the mask problem for another week or two (on vacation now), but if your change fixes the problem for you, I'm sure it's fine (I just noticed that some results on astro-lsst-01 still have the problem and realized I hadn't submitted a pull request a month ago when I fixed it locally).

rhiannonlynne commented:
The existing plots on astro-lsst-01 will probably continue to have the same problem because we don't usually replace them -- but anything new should not. I think that anything on the 8081 port should not have the problem, as these were created after the fix.
